### PR TITLE
Implement date navigation links in process monitoring screen (#6241)

### LINKED ISSUE #6241: Implement date navigation links in process monitoring screen
We aim to enhance the user experience on the process monitoring screen by adding date navigation links at the top-right corner of the screen. The links will be as follows:

<< Year Month Week >>

Each link will have the following functionality:

- [x] - Year: Sets the date values in the form to the current year.
- [x] - Month: Sets the date values in the form to the current month.
- [x] - Week: Sets the date values in the form to the current week.
- [x] - '<<': Moves to the past based on the previously selected period (year, month, or week).
- [x] - '>>': Moves to the future based on the previously selected period.

- [x] By default, upon landing on the screen, the date will be set to the current year, and the startDate and endDate fields will be adjusted accordingly.

- [x] If the user manually changes the date and then clicks on <<, the dates should shift as follows:

- New StartDate = current StartDate - (current EndDate - current StartDate)
- New EndDate = current EndDate - (current EndDate - current StartDate)

- [x] If the startDate and endDate fields are not set (i.e., the user has cleared the date fields), no action should be taken.

- [ ] If only the startDate or endDate is set, the date should shift based on the last selected period (year, month, or week).

- [x] When the user clicks on ">>", the system should advance to future dates, mirroring the behavior of the "<<" button which navigates to past dates.

- [x] These rules should be implemented in ui/main/src/app/business/view/processmonitoring/processmonitoring.view.ts with corresponding unit tests. For instance, a method named `getDatesAfterPeriodClick(periodClicked: string) `could be created, which returns a new object containing startDate and endDate.

- [x] For end-to-end testing, a single use case should be covered using Cypress. For example, clicking on "week" and verifying that the dates change accordingly.

### PR DESCRIPTION
**Pull request by vlo-rte**
- In release note :
  -  In chapter :  Features
  -  Text : #6241 : Implement Date Navigation Links in Process Monitoring Screen

### CODE REVIEW

File: ui/main/src/app/modules/processmonitoring/processmonitoring.component.scss [hunk 1]
```diff
@@ -77,6 +77,61 @@
...
+                background-color: var(--opfab-bgcolor);
+                color: var(--opfab-navbar-color);
+                text-decoration: none;
+                cursor: pointer;
+            }
+
+            .circle-btn {
+                border-radius: 50%;
+                border: none;
+            }
```
Comment (freddidierRTE) [inline ui/main/src/app/modules/processmonitoring/processmonitoring.component.scss]: i do not think "circle-btn" it is used

File: ui/main/src/app/modules/processmonitoring/processmonitoring.component.ts [hunk 2]
```diff
@@ -183,6 +187,43 @@ export class ProcessMonitoringComponent implements OnDestroy, OnInit, AfterViewI
...
+                    this.periodClicked
+                );
+            } else {
+                newDates = this.processMonitoringView.getDatesWhenMoving(
+                    this.processMonitoringForm.value.activeFrom,
+                    this.processMonitoringForm.value.activeTo,
+                    isForward
+                );
+            }
+            this.processMonitoringForm.patchValue({activeFrom: newDates.activeFrom, activeTo: newDates.activeTo});
```
Comment (freddidierRTE) [inline ui/main/src/app/modules/processmonitoring/processmonitoring.component.ts]: How about grouping in one method getDatesWhenMoving with periodClicked and if periodClicked is set to empty it calls internally getDatesWhenMovingWithPeriodClicked ?

File: ui/main/src/app/modules/processmonitoring/processmonitoring.component.ts [hunk 3]
```diff
@@ -257,6 +306,7 @@ export class ProcessMonitoringComponent implements OnDestroy, OnInit, AfterViewI
         this.columnFilters = [];
         this.firstQueryHasBeenDone = false;
         this.setDateFilterBounds();
+        this.resetPeriodClicked();
```
Comment (freddidierRTE) [inline ui/main/src/app/modules/processmonitoring/processmonitoring.component.ts:296]: When reset form it should go back to initial state which is year mode with current year selected

File: ui/main/src/app/business/view/processmonitoring/processmonitoring.view.ts [hunk 4]
```diff
@@ -103,4 +104,191 @@ export class ProcessMonitoringView {
...
+                        ? activeToDateFormat.getFullYear() - 1 + '-12-01T00:00'
+                        : activeToDateFormat.getFullYear() +
+                          '-' +
+                          String(activeToDateFormat.getMonth()).padStart(2, '0') +
+                          '-01T00:00'
+            };
+        }
+    }
+
+    public getDatesWhenMovingWithWeekPeriodClicked(
```
Comment (freddidierRTE) [inline ui/main/src/app/business/view/processmonitoring/processmonitoring.view.ts]: To set as private as it is only used internally

File: ui/main/src/app/business/view/processmonitoring/processmonitoring.view.ts [hunk 5]
```diff
@@ -103,4 +104,191 @@ export class ProcessMonitoringView {
...
+            };
+        } else {
+            return {
+                activeFrom: activeFromDateFormat.getFullYear() - 1 + '-01-01T00:00',
+                activeTo: activeToDateFormat.getFullYear() - 1 + '-01-01T00:00'
+            };
+        }
+    }
+
+    public getDatesWhenMovingWithMonthPeriodClicked(
```
Comment (freddidierRTE) [inline ui/main/src/app/business/view/processmonitoring/processmonitoring.view.ts]: To set as private as it is only used internally

File: ui/main/src/app/business/view/processmonitoring/processmonitoring.view.ts [hunk 6]
```diff
@@ -103,4 +104,191 @@ export class ProcessMonitoringView {
...
+        if (periodClicked === 'month') {
+            return this.getDatesWhenMovingWithMonthPeriodClicked(activeFrom, activeTo, isForward);
+        }
+        if (periodClicked === 'week') {
+            return this.getDatesWhenMovingWithWeekPeriodClicked(activeFrom, activeTo, isForward);
+        }
+        return null;
+    }
+
+    public getDatesWhenMovingWithYearPeriodClicked(
```
Comment (freddidierRTE) [inline ui/main/src/app/business/view/processmonitoring/processmonitoring.view.ts]: To set as private as it is only used internally

File: ui/main/src/app/business/view/processmonitoring/processmonitoring.view.ts [hunk 7]
```diff
@@ -103,4 +104,191 @@ export class ProcessMonitoringView {
...
+                activeToDateFormat.getTime() - (activeToDateFormat.getTime() - activeFromDateFormat.getTime())
+            ).toLocaleString('fr-FR');
+        }
+        return {
+            activeFrom: this.formatDateLocalStringForInput(newActiveFromDate),
+            activeTo: this.formatDateLocalStringForInput(newActiveToDate)
+        };
+    }
+
+    public formatDateLocalStringForInput(dateLocalString: string) {
```
Comment (freddidierRTE) [inline ui/main/src/app/business/view/processmonitoring/processmonitoring.view.ts]: To set as private as it is only used internally

File: ui/main/src/app/business/view/processmonitoring/processmonitoring.view.ts [hunk 8]
```diff
@@ -103,4 +104,191 @@ export class ProcessMonitoringView {
...
+            newActiveToDate = new Date(
+                activeToDateFormat.getTime() + (activeToDateFormat.getTime() - activeFromDateFormat.getTime())
+            ).toLocaleString('fr-FR');
+        } else {
+            newActiveFromDate = new Date(
+                activeFromDateFormat.getTime() - (activeToDateFormat.getTime() - activeFromDateFormat.getTime())
+            ).toLocaleString('fr-FR');
+            newActiveToDate = new Date(
+                activeToDateFormat.getTime() - (activeToDateFormat.getTime() - activeFromDateFormat.getTime())
+            ).toLocaleString('fr-FR');
```
Comment (freddidierRTE) [inline ui/main/src/app/business/view/processmonitoring/processmonitoring.view.ts]: Passing via the fr-FR format seems a little weird 

Could you just keep in this method a date object and in formatDateLocalStringForInput directly convert to the wanted format using for example moment or getYear(), get Month() ... to be consistant with other methods ?

File: ui/main/src/app/business/view/processmonitoring/processmonitoring.view.ts [hunk 9]
```diff
@@ -103,4 +104,191 @@ export class ProcessMonitoringView {
...
+
+    public formatDateLocalStringForInput(dateLocalString: string) {
+        return (
+            dateLocalString.substring(6, 10) +
+            '-' +
+            dateLocalString.substring(3, 5) +
+            '-' +
+            dateLocalString.substring(0, 2) +
+            'T' +
+            dateLocalString.substring(11, 20)
```
Comment (freddidierRTE) [inline ui/main/src/app/business/view/processmonitoring/processmonitoring.view.ts]: The format seems not consistent with other methods , the substring (11,20) leads to add seconds in the string where in other method you do not have  (YYYY-MM-DDTHH:MM)

File: ui/main/src/app/modules/processmonitoring/processmonitoring.component.scss [hunk 10]
```diff
@@ -77,6 +77,61 @@
...
+        justify-content: flex-end;
+
+        li {
+            padding-left: 0px;
+            padding-right: 25px;
+            // The following 3 lines avoid having the text go an a new line
+            // Arise only on chrome (OC-1563)
+            text-overflow: clip;
+            white-space: nowrap;
+            overflow: hidden;
```
Comment (freddidierRTE) [inline ui/main/src/app/modules/processmonitoring/processmonitoring.component.scss]: It seems it is not needed in this use case (i tested locally with ng serve mode)

**Review by freddidierRTE — CHANGES_REQUESTED** [review]

File: ui/main/src/app/modules/processmonitoring/processmonitoring.component.scss (hunk 1 above)
Comment (vlo-rte) [reply ui/main/src/app/modules/processmonitoring/processmonitoring.component.scss]: Done.

File: ui/main/src/app/business/view/processmonitoring/processmonitoring.view.ts (hunk 4 above)
Comment (vlo-rte) [reply ui/main/src/app/business/view/processmonitoring/processmonitoring.view.ts]: Done.

File: ui/main/src/app/business/view/processmonitoring/processmonitoring.view.ts (hunk 5 above)
Comment (vlo-rte) [reply ui/main/src/app/business/view/processmonitoring/processmonitoring.view.ts]: Done.

File: ui/main/src/app/business/view/processmonitoring/processmonitoring.view.ts (hunk 6 above)
Comment (vlo-rte) [reply ui/main/src/app/business/view/processmonitoring/processmonitoring.view.ts]: Done.

File: ui/main/src/app/business/view/processmonitoring/processmonitoring.view.ts (hunk 7 above)
Comment (vlo-rte) [reply ui/main/src/app/business/view/processmonitoring/processmonitoring.view.ts]: Done.

File: ui/main/src/app/modules/processmonitoring/processmonitoring.component.scss (hunk 10 above)
Comment (vlo-rte) [reply ui/main/src/app/modules/processmonitoring/processmonitoring.component.scss]: Done.

File: ui/main/src/app/modules/processmonitoring/processmonitoring.component.ts (hunk 2 above)
Comment (vlo-rte) [reply ui/main/src/app/modules/processmonitoring/processmonitoring.component.ts]: I guess you mean "and if periodClicked is not empty".
Done.

File: ui/main/src/app/business/view/processmonitoring/processmonitoring.view.ts (hunk 9 above)
Comment (vlo-rte) [reply ui/main/src/app/business/view/processmonitoring/processmonitoring.view.ts]: I guess you mean "you have (YYYY-MM-DDTHH:MM)"

Comment (vlo-rte) [reply ui/main/src/app/business/view/processmonitoring/processmonitoring.view.ts]: Done.

File: ui/main/src/app/modules/processmonitoring/processmonitoring.component.ts (hunk 3 above)
Comment (vlo-rte) [reply ui/main/src/app/modules/processmonitoring/processmonitoring.component.ts:296]: Done.

File: ui/main/src/app/business/view/processmonitoring/processmonitoring.view.ts (hunk 8 above)
Comment (vlo-rte) [reply ui/main/src/app/business/view/processmonitoring/processmonitoring.view.ts]: Done.

**Review by freddidierRTE — APPROVED** [review]

**Pull request merged**